index.bs

```diff
@@ -1462,3 +1484,231 @@ The [=remote end steps=] with |command parameters| are:
     1. Return the result of [=updating the event map=] with [=current session=],
        |list of event names|, |list of contexts| and enabled false.
 </div>
+
+
+## browsingContext ## {#module-browsingContext}
+
+The <dfn export for=modules>Browsing Context</dfn> module contains commands and
+events relating to browsing contexts.
+
+### Definition ### {#module-browsingContext-definition}
+
+[=remote end definition=]
+
+<pre class="cddl remote-cddl">
+
+BrowsingContextCommand = (BrowsingContextGetAllCommand)
+</pre>
+
+[=local end definition=]
+
+<pre class="cddl local-cddl">
+
+BrowsingContextResult = (BrowsingContextGetAllResult)
+
+BrowsingContextEvent = (
+    BrowsingContextCreatedEvent //
+    BrowsingContextDestroyedEvent
+)
+
+BrowsingContextInfo = {
+  context: BrowsingContext,
+  parent: BrowsingContext / null,
+  url: text,
+}
+
+</pre>
+
+### Types ### {#module-browsingcontext-types}
+
+#### BrowsingContext #### {#types-browsingcontext}
+
+[=remote end definition=] and [=local end definition=]
+
+<pre class="cddl remote-cddl local-cddl">
+BrowsingContext = text;
+</pre>
+
+Each [=/browsing context=] has an associated <dfn export>browsing context
+id</dfn>, which is a string uniquely identifying that browsing context. This is
+implicitly set when the context is created. For browsing contexts with an
+associated WebDriver [=window handle=] the [=/browsing context id=] must be the
+same as the [=window handle=].
+
+<div algorithm>
+To <dfn>get a browsing context</dfn> given |context id|:
+
+ 1. If |context id| is null, return [=success=] with data null.
+
+ 1. If there is no browsing context with [=browsing context id=] |context id| return
+    [=error=] with [=error code=] [=no such frame=]
+
+ 1. Let |context| be the browsing context with id |context id|.
+
+ 1. Return [=success=] with data |context|
+
+</div>
+
+<div algorithm>
+To <dfn>get the browsing context info</dfn> given |context|:
+ 1. Let |context id| be the [=browsing context id=] for |context|.
+
+ 1. If |context| has a [=parent browsing context=] let |parent id| be the
+    [=browsing context id=] of that parent. Otherwise let |parent id| be null.
+
+ 1. Let |document| be |context|'s [=active document=].
+
+ 1. Let |url| be the result of running the [=URL serializer=], given
+    |document|'s [=spec=html/URL=].
+
+ 1. Let |context info| be a map matching the <code>BrowsingContextInfo</code>
+    production with the <code>context</code> field set to |context id|, the
+    <code>parent</code> field set to |parent id|, and the <code>url</code> field
+    set to |url|.
+
+ 1. Return |context info|.
+
+</div>
+
+### Commands ### {#module-browsingContext-commands}
+
+#### list ####  {#command-browsingContext-list}
+
+The <dfn export for=commands>browsingContext.list</dfn> command returns a list
```

Comment:
I see that CDP's equivalent for this lists the whole browsing context tree as one command:
https://chromedevtools.github.io/devtools-protocol/tot/Page/#method-getFrameTree

I believe that's actually better, because there's no risk of getting an inconsistent view of the tree by trying to build it up over multiple roundtrips.

How does this work in Gecko's protocol(s)?

Reply:
I made this recurse and added a `maxDepth` parameter in case we want to allow getting just top-level frames for example. I can remove that if we think it's overkill, but it's not much complexity in the spec.